web/web.go

```diff
@@ -539,6 +539,11 @@ func (h *Handler) Run(ctx context.Context) error {
 		conntrack.TrackWithName("http"),
 		conntrack.TrackWithTracing())
 
+	return listener, nil
+}
+
+// Run serves the HTTP endpoints.
+func (h *Handler) Run(ctx context.Context, listener net.Listener) error {
```

Comment:
should we check for nil listener and return an error ? Or call Listener() in that case?